### PR TITLE
`autoPay` mutation endpoints

### PR DESCRIPTION
**Pull request by caillou**
* Adds a resolver for `membership.autoPayIsMutable`
* Adds a `setMembershipAutoPay` mutation

BREAKING CHANGE:

Removes the following GraphQL mutations:

* disableMembershipAutoPay
* enableMembershipAutoPay

### CODE REVIEW

**Comment by patrickvenetz** [comment]
giphy (1)

File: servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/setMembershipAutoPay.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const { pgdb, req } = context
+  Roles.ensureUserHasRole(req.user, 'supporter')
+
+  const transaction = await pgdb.transactionBegin()
+
+  try {
+    const membership = await pgdb.public.memberships.findOne({
+      id: id
+    })
+    const autoPayIsMutable = await autoPayIsMutableResolver(membership, null, context)
```
Comment (patte) [inline servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/setMembershipAutoPay.js:15]:
```suggestion
    const autoPayIsMutable = await autoPayIsMutableResolver(membership, null, { ...context, pgdb: transaction })
```

File: servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/setMembershipAutoPay.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    if (!autoPayIsMutable) {
+      throw new Error(
+        `The autoPay field of the membership ${membership.id} is not mutable.`
```
Comment (patte) [inline servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/setMembershipAutoPay.js:19]: please add to translation file

File: servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/setMembershipAutoPay.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      return membership
+    }
+
+    await pgdb.public.memberships.updateOne({
+      id
+    }, {
+      autoPay
+    })
+
+    membership.autoPay = autoPay
```
Comment (patte) [inline servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/setMembershipAutoPay.js:33]:
```suggestion
    const updatedMembership = await transaction.public.memberships.updateAndGetOne({
      id
    }, {
      autoPay
    })
```

File: servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/setMembershipAutoPay.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    await transaction.transactionCommit()
+
+    await createCache({
+      prefix: `User:${membership.userId}`
+    }, context).invalidate()
+
+    return membership
```
Comment (patte) [inline servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/setMembershipAutoPay.js:41]:
```suggestion
    return updatedMembership
```

**Comment by caillou** [comment]
@patte Thanks for your review! I updated the PR. Could you have another quick look?

**Pull request merged**